src/cascadia/TerminalApp/App.cpp

```diff
@@ -511,13 +651,20 @@ namespace winrt::TerminalApp::implementation
         {
             auto newSettings = CascadiaSettings::LoadAll(saveOnLoad);
             _settings = std::move(newSettings);
-            hr = S_OK;
+            const auto& warnings = _settings->GetLoadWarnings();
+            hr = warnings.size() == 0 ? S_OK : S_FALSE;
         }
         catch (const winrt::hresult_error& e)
         {
             hr = e.code();
+            _settingsLoadExceptionText = e.message();
             LOG_HR(hr);
         }
+        catch (const ::TerminalApp::SettingsLoadErrors error)

```

Comment:
whoa whoa whoa i don't know if we should _catch an enum type_.

Reply:
This seems to work fine.... https://stackoverflow.com/a/20951192/1481137

Reply:
this clashes with https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Re-exception-types

But also it is tricky to throw something which doesn't inherit from std::exception or another base class as it risks the exception to bubble past all catch statements and crash the program.

Reply:
Yeah, even though it's possible we shouldn't necessarily _do_ it. :smile:

Reply:
we could allocate an hresult domain!

Reply:
and throw an hresult_exception!